lib/currentCoupons.ts

```diff
@@ -0,0 +1,22 @@
+/*
+ * Copyright (c) 2019-2024 Bjoern Kimminich & the OWASP Juice Shop contributors.
+ * SPDX-License-Identifier: MIT
+ */
+
+import logger from './logger.ts'
+import colors from 'colors'
+import request from 'sync-request'
+
+export default (apiEndpoint: string = 'https://5j4d1u7jhf.execute-api.eu-west-1.amazonaws.com/default/JuicyCouponFunc'): { expiryDate: string, discountCodes: Record<string, string> } => {
```

Comment:
i think this might be a good point / reason to remove the sync-request lib and switch to the now build in fetch function in node.js

having to manually overwrite the function here really isn't great from a maintainability standpoint.
Also would allow us to remove one more external dependency.